src/validation.cpp

```diff
@@ -5222,3 +5222,13 @@ std::optional<int> ChainstateManager::GetSnapshotHeight()
     }
     return (*base)->nHeight;
 }
+
+bool ChainstateManager::IsAnyChainInIBD()
```

Comment:
In commit "p2p: don't advertise until we finish all IBDs" (8dc65c32ffb533402bc0c82ccf3f1fe5535df521)

It seems messy for individual chains have their own IBD states, and interact with shared `fImporting` and `fReindex` and `nMinimumChainWork` globals, and use their own latches.

I think the IBD logic would be more straightforward if the `CChainState::IsInitialBlockDownload` method and `CChainState::m_cached_finished_ibd` member were moved from the `CChainState` class to the `ChainstateManager` class and consolidated in one place.